src/shared/cloudformation/templateRegistry.ts

```diff
@@ -125,18 +128,101 @@ export class CloudFormationTemplateRegistry {
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param unfilteredTemplates Array containing TemplateDatum objects to filter
  */
-export function getResourcesFromTemplateDatum(templateDatum: TemplateDatum): Map<string, CloudFormation.Resource> {
-    const map = new Map<string, CloudFormation.Resource>()
-    for (const resourceKey of Object.keys(templateDatum.template.Resources!)) {
-        const resource = templateDatum.template.Resources![resourceKey]
-        if (resource) {
-            map.set(resourceKey, resource)
+export function getTemplatesAssociatedWithHandler(
+    filepath: string,
+    handler: string,
+    unfilteredTemplates: TemplateDatum[] = CloudFormationTemplateRegistry.getRegistry().registeredTemplates
+): TemplateDatum[] {
+    // find potential matching templates
+    return unfilteredTemplates.filter(templateDatum => {
+        return !!getResourceAssociatedWithHandlerFromTemplateDatum(filepath, handler, templateDatum)
+    })
+}
+
+/**
+ * Returns the first Cloudformation Resource in a TemplateDatum that is tied to the filepath and handler given.
+ * Returns undefined if none are found.
+ * TODO: Is there a reason to return more than the first resource?
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param templateDatum TemplateDatum object to search through
+ */
+export function getResourceAssociatedWithHandlerFromTemplateDatum(
+    filepath: string,
+    handler: string,
+    templateDatum: TemplateDatum
+): CloudFormation.Resource | undefined {
+    const templateDirname = path_.dirname(templateDatum.path)
+    // template isn't a parent or sibling of file
+    if (!isInDirectory(templateDirname, path_.dirname(filepath))) {
+        return undefined
+    }
+
+    const resources = templateDatum.template.Resources
+    if (!resources) {
+        return undefined
+    }
+
+    for (const key of Object.keys(resources)) {
+        const resource = resources[key]
+        // check if some sort of serverless function
+        if (
+            resource &&
+            [CloudFormation.SERVERLESS_FUNCTION_TYPE, CloudFormation.LAMBDA_FUNCTION_TYPE].includes(resource.Type)
+        ) {
+            const registeredRuntime = CloudFormation.getStringForProperty(
+                resource.Properties?.Runtime,
+                templateDatum.template
+            )
+            const registeredCodeUri = CloudFormation.getStringForProperty(
+                resource.Properties?.CodeUri,
+                templateDatum.template
+            )
+            const registeredHandler = CloudFormation.getStringForProperty(
+                resource.Properties?.Handler,
+                templateDatum.template
+            )
+
+            if (registeredRuntime && registeredHandler && registeredCodeUri) {
+                if (dotNetRuntimes.includes(registeredRuntime)) {
```

Comment:
I can't recall offhand what the handler formats look like for each of the runtime cases below. It could help reduce cognitive load when looking at this code if a comment was added in the if/else scopes with an example of a matching handler and filepath.

Reply:
Can do. All interpreted languages are the same, at this time C# is the only exception. I'm not super knowledgeable about Omnisharp but we might be able to get more accurate recommendations based on that.